src/swarm-scenario/Swarm/Game/Scenario/Topography/WorldPalette.hs

```diff
@@ -71,11 +69,13 @@ constructWorldMap mappedPairs maskChar =
  where
   invertedMappedPairs = map (swap . fmap toKey) mappedPairs
 
+  renderMapCell :: Maybe CellPaintDisplay -> Char
   renderMapCell maybeC = case maybeC of
     Nothing -> maskChar
     Just c ->
-      -- NOTE: This lookup should never fail
-      M.findWithDefault (error "Palette lookup failed!") k $
+      -- NOTE: This lookup should never fail; if it does for some
+      -- reason, silently return a space rather than crash the game
+      M.findWithDefault ' ' k $
```

Comment:
Perhaps make this a visible character like 'Z', just in case we we actually do encounter it at runtime and need to search for it in the source code.

Reply:
Good idea, done.